### PR TITLE
fix: assertion failure when requiring missing module in autocmd

### LINKED ISSUE #15348: Assertion failure when requiring missing lua module, in a Cmdline* autocmd, when package.path is long
### Neovim Version

NVIM v0.8.0-dev-961-g0b72e23bf (Debug build)

### Operating system/version:

Linux/Arch

### Terminal name/version:

NA

### TERM environment variable

NA

### Installation

asdf

### Steps to reproduce

This can occur if you're using packer, since it sets a longer than usual `package.path`.

**init.lua**

```lua
-- value pulled from packer_compiled.lua source
local package_path_str = "/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/?.lua;/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/?/init.lua;/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/?.lua;/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/?/init.lua"
package.path = package.path .. ';' .. package_path_str

vim.api.nvim_exec(
  [[
  augroup Crasher
    autocmd!
    autocmd CmdlineLeave / lua require("fake-moustache")
  augroup end
]],
  false
)
```

`Containerfile` reproduction against master HEAD:

<details>

```containerfile
# Build neovim separately in the first stage
FROM alpine:latest AS base

RUN apk --no-cache add \
    autoconf \
    automake \
    build-base \
    cmake \
    ninja \
    coreutils \
    curl \
    gettext-tiny-dev \
    git \
    libtool \
    pkgconf \
    unzip

# Build neovim (and use it as an example codebase
RUN git clone https://github.com/neovim/neovim.git

ARG VERSION=master
RUN cd neovim \
    && git checkout ${VERSION} \
    && make CMAKE_BUILD_TYPE=RelWithDebInfo install

RUN echo $' \n\
-- value pulled from packer_compiled.lua source \n\
local package_path_str = "/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/?.lua;/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/?/init.lua;/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/?.lua;/home/username/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/?/init.lua" \n\
package.path = package.path .. ";" .. package_path_str \n\
vim.api.nvim_exec( \n\
  [[ \n\
  augroup Crasher \n\
    autocmd! \n\
    autocmd CmdlineLeave / lua require("fake-moustache") \n\
  augroup end \n\
]], \n\
  false \n\
)' >> /init.lua

CMD ["nvim", "--clean", "-u", "/init.lua"]
```

</details>

**Steps**

1) `nvim --clean -u init.lua`
2) `/<cr>`

Note we:

- have a long `package.path`
- require a missing module in an `autocmd` (please remove your fake moustache if you have one)

If you comment out setting `package.path`, the `autocmd` will simply print an error.

### Vim

Does not crash, prints normal lua "require error"

<details>

```viml
lua package.path = string.rep((package.path .. ";"), 20)

augroup Crasher
  autocmd!
  autocmd CmdlineLeave / lua require("fake-moustache")
augroup end
```

</details>

### Expected behavior

Print error to messages.

### Actual behavior

```
Assertion failed: len < 0 || memchr(str, 0, (size_t)len) == NULL (/neovim/src/nvim/message.c: msg_puts_attr_len: 2058)
                         Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by dundargoc**
Closes https://github.com/neovim/neovim/issues/15348

### CODE REVIEW

File: src/nvim/ex_getln.c [hunk 1]
```diff
@@ -987,7 +987,8 @@ static uint8_t *command_line_enter(int firstc, long count, int indent, bool init
 
   if (!tl_ret && ERROR_SET(&err)) {
     msg_putchar('\n');
-    msg_printf_attr(HL_ATTR(HLF_E)|MSG_HIST, (char *)e_autocmd_err, err.msg);
+    emsg(e_autocmd_err);
+    msg(err.msg);
```
Comment (bfredl) [inline src/nvim/ex_getln.c:991]: maybe `emsg` so the rest also printed with `ErrorMsg` color

Comment (justinmk) [reply src/nvim/ex_getln.c]: `e_autocmd_err` has a `%s` in it, so it expects an argument. How about `semsg` ?:

    semsg(e_autocmd_err, err.msg);

File: test/functional/autocmd/cmdline_spec.lua [hunk 2]
```diff
@@ -78,56 +78,56 @@ describe('cmdline autocommands', function()
...
     feed("put ='lorem ipsum'<cr>")
-    screen:expect([[
-                                                                              |
+    screen:expect{grid=[[
       {4:                                                                        }|
       :                                                                       |
       {2:E5500: autocmd has thrown an exception: Vim(echoerr):FAIL}               |
       :put ='lorem ipsum'                                                     |
-      {2:E5500: autocmd has thrown an exception: Vim(echoerr):very error}         |
+      {2:E5500: autocmd has thrown an exception: %s}                              |
```
Comment (justinmk) [inline test/functional/autocmd/cmdline_spec.lua:86]: literal `%s` is not wanted. with `semsg` this test should need less changes.

Comment (dundargoc) [reply test/functional/autocmd/cmdline_spec.lua:86]: Doh, now it makes sense. Thanks.

File: test/functional/autocmd/cmdline_spec.lua [hunk 3]
```diff
@@ -92,8 +92,8 @@ describe('cmdline autocommands', function()
     -- cmdline was still executed
     feed('<cr>')
     screen:expect([[
-                                                                              |
-      ^lorem ipsum                                                             |
+      ^                                                                        |
+      {1:~                                                                       }|
```
Comment (justinmk) [inline test/functional/autocmd/cmdline_spec.lua:96]: why isn't "lorem ipsum" displayed now?

**Comment by clason** [comment]
Is this still needed?

File: test/functional/autocmd/cmdline_spec.lua (hunk 3 above)
Comment (dundargoc) [reply test/functional/autocmd/cmdline_spec.lua:96]: I don't know, I thought `semsg` and `msg_printf_attr` just printed a messages, so not sure why there's a functional difference in behavior.

Comment (dundargoc) [reply test/functional/autocmd/cmdline_spec.lua]: OK, I think the difference is that `semsg` cancels current command while `msg_printf_attr` doesn't.

**Comment by clason** [comment]
@bfredl looks good now?

**Pull request merged**